File: 9_classUser/9.cpp
#include <iostream>
#include <cstdlib>
#include <cstdio>
#include <string>

using namespace std;

class User {
private:
    string nombre;
    string apellido;
    int edad;
    bool isAdmin;
public:
    User();
    User(string, string, int, bool);
    string getNombre();
    string getApellido();
    int getEdad();
    bool isAdministrador();
    void setNombre(string nombre);
    void setApellido(string apellido);
    void setEdad(int edad);
    void setIsAdmin(bool isAdmin);
    string toString();
    virtual ~User();
};

User::User() {
    this->nombre = "desconocido";
    this->apellido = "desconocido";
    this->edad = -1;
    this->isAdmin = false;
}

User::User(string nombre, string apellido, int edad, bool isAdmin) {
    this->nombre = nombre;
    this->apellido = apellido;
    this->edad = edad;
    this->isAdmin = isAdmin;
}

string User::getNombre() {
    return this->nombre;
}

string User::getApellido() {
    return this->apellido;
}

int User::getEdad() {
    return this->edad;
}

bool User::isAdministrador() {
    return this->isAdmin;
}

void User::setNombre(string nombre) {
    this->nombre = nombre;
}

void User::setApellido(string apellido) {
    this->apellido = apellido;
}

void User::setEdad(int edad) {
    this->edad = edad;
}

void User::setIsAdmin(bool isAdmin) {
    this->isAdmin = isAdmin;
}

string User::toString() {
    return
        this->nombre + " " + this->apellido + " " + to_string(this->edad) + " " + to_string(this->isAdmin);
}

User::~User() {
    this->nombre = "";
    this->apellido = "";
    this->edad = 0;
    this->isAdmin = false;
}

int main() {
    User u("David", "Betancourt", 20, true);
    u.setEdad(21);
    cout << u.toString() << endl;

    return 0;
}

File: 9_ClassUser/9.rb
class User
    def initialize()
        @nombre = "desconocido"
        @apellido = "desconocido"
        @edad = -1
        @isAdmin = false
    end

    def initialize(nombre, apellido, edad, isAdmin)
        @nombre = nombre
        @apellido = apellido
        @edad = edad
        @isAdmin = isAdmin
    end

    def nombre
        @nombre
    end

    def nombre=(nombre)
        @nombre = nombre
    end

    def apellido
        @apellido
    end

    def apellido=(apellido)
        @apellido = apellido
    end

    def edad
        @edad
    end

    def edad=(edad)
        @edad = edad
    end

    def isAdmin
        @isAdmin
    end

    def isAdmin=(isAdmin)
        @isAdmin = isAdmin
    end

    def toString
        return @nombre + " " + @apellido + " " + @edad.to_s + " " + @isAdmin.to_s
    end
end

usuario = User.new("David", "Betancourt", 20, true)
print "Escribe el nombre: "
usuario.nombre = gets.chomp.to_s
usuario.edad = usuario.edad + 1
puts usuario.toString
